x/rollapp/keeper/hooks.go

```diff
@@ -0,0 +1,40 @@
+package keeper
+
+import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	epochstypes "github.com/osmosis-labs/osmosis/v15/x/epochs/types"
+)
+
+var _ epochstypes.EpochHooks = epochHooks{}
+
+const defaultDeleteStateInfoTimestampThreshold = time.Hour * 24
+
+type epochHooks struct {
+	Keeper
+}
+
+func (k Keeper) GetEpochHooks() epochstypes.EpochHooks {
+	return epochHooks{
+		Keeper: k,
+	}
+}
+
+// AfterEpochEnd is the epoch end hook.
+// We want to clean up the demand orders that are with underlying packet status which are finalized.
+func (e epochHooks) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, _ int64) error {
+	if epochIdentifier != e.EpochIdentifier(ctx) {
+		return nil
+	}
+
+	currentTimestamp := ctx.BlockTime()
+	seqUnbondingTime := e.sequencerKeeper.UnbondingTime(ctx)
+	endTimestamp := currentTimestamp.Add(-seqUnbondingTime).Add(-defaultDeleteStateInfoTimestampThreshold) // add a threshold for good measure
```

Comment:
good measure is not a technical term

Reply:
what is the reason
we're not baking cakes

Reply:
yeah, there was much debate whether this was necessary. Tbh I don't see it as much more than just "for good measure", so I wrote it like that. 
I'm still considering if that threshold can be removed, as I don't see much utility in it
@omritoptix wdyt?